scalyr_agent/profiler.py

```diff
@@ -106,86 +236,215 @@ def _start(self, config, current_time):
             yappi.set_clock_type(self._profile_clock)
         global_log.log(
             scalyr_logging.DEBUG_LEVEL_0,
-            "Starting profiling using '%s' clock. Duration: %d seconds"
+            "Starting CPU profiling using '%s' clock. Duration: %d seconds"
             % (self._profile_clock, self._profile_end - self._profile_start),
         )
         yappi.start()
 
     def _stop(self, config, current_time):
         yappi.stop()
-        global_log.log(scalyr_logging.DEBUG_LEVEL_0, "Stopping profiling")
+        global_log.log(scalyr_logging.DEBUG_LEVEL_0, "Stopping CPU profiling")
         stats = yappi.get_func_stats()
-        path = os.path.join(config.agent_log_path, config.profile_log_name)
-        if os.path.exists(path):
-            os.remove(path)
+        if os.path.exists(self._data_file_path):
+            os.remove(self._data_file_path)
 
         # pylint bug https://github.com/PyCQA/pylint/labels/topic-inference
-        stats.save(path, "callgrind")  # pylint: disable=no-member
+        stats.save(self._data_file_path, "callgrind")  # pylint: disable=no-member
 
         lines = 0
 
         # count the lines
-        f = open(path)
+        f = open(self._data_file_path)
         try:
             for line in f:
                 lines += 1
         finally:
             f.close()
 
         # write a status message to make it easy to find the end of each profile session
-        f = open(path, "a")
+        f = open(self._data_file_path, "a")
         try:
             f.write(
                 "\n# %s, %s clock, total lines: %d\n"
-                % (path, self._profile_clock, lines)
+                % (self._data_file_path, self._profile_clock, lines)
             )
         finally:
             f.close()
 
         yappi.clear_stats()
         del stats
 
-    def update(self, config, current_time=None):
+        global_log.log(
+            scalyr_logging.DEBUG_LEVEL_0,
+            "CPU profiling data written to %s",
+            self._data_file_path,
+        )
+
+
+class PeriodicMemorySummaryCaptureThread(Thread):
```

Comment:
You might want to use our `StoppableThread` here.  The advantage is that it has a sleep method that will exit early if you stop the thread.  Down below, this thread could take up to 5 seconds to stop.

It's not clear to me if we have folks that are blocking on that stop or not.  For example, if the agent is doing a soft-restart due to a configuration change, do we tell any of the profiling code to stop before we loading the new configuration?

Reply:
Ah, cool, I missed the existing ``StoppableThread`` abstraction.

I will update the code to use it. I don't think we will save any lines of code, but it's still a good idea to use the existing abstraction which pretty much does the same thing.

Reply:
Pushed a change to use existing StoppableThread abstraction - 06fd81d.